core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -395,7 +395,9 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       String oldLocation = base == null ? null : base.metadataFileLocation();
 
       synchronized (InMemoryCatalog.this) {
-        if (null == base && !namespaceExists(tableIdentifier.namespace())) {
+        if (null == base
+            && !namespaceExists(tableIdentifier.namespace())
+            && !tableIdentifier.namespace().isEmpty()) {
```

Comment:
Why do we need to check for empty here? Shouldn't this be consistent with `namespaceExists` so that method covers it?

Reply:
that's similar to how it's done in https://github.com/apache/iceberg/blob/2446cee5cf0ad93a2be9a68f0b2f7f6fa6edb865/core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java#L141, where the empty namespace is allowed for listing

Reply:
I've reverted this change as it should actually require that the empty namespace has been created first